cabal-install/src/Distribution/Client/CmdExec.hs

```diff
@@ -163,9 +170,15 @@ execAction flags@NixStyleFlags{..} extraArgs globalFlags = do
 
   -- Some dependencies may have executables. Let's put those on the PATH.
   extraPaths <- pathAdditions verbosity baseCtx buildCtx
+  let configProgPathExtras =
+        fromNubList
+          . projectConfigProgPathExtra
+          . projectConfigShared
+          . projectConfig
+          $ baseCtx
```

Comment:
I think that is the case yeah. I also think this is transported into a different structure somewhere along the path of `CmdBuild` et al, because it works in that command. This is part of my confusion, I don't know exactly what is `baseCtx` then `targetCtx` etc etc.